gdx/test/com/badlogic/gdx/math/FloatCounterTest.java

```diff
@@ -0,0 +1,152 @@
+		Assert.assertEquals(-5.2f, counter1.max, 0);
+		counter1.put(12.0f);
+		Assert.assertEquals(-14.0f, counter1.total, 0);
+		Assert.assertEquals(6, counter1.count);
+		Assert.assertEquals(12.0f, counter1.latest, 0);
+		Assert.assertEquals(-2.333333333f, counter1.average, 0.000000001f);
+		Assert.assertEquals(-3.0f, counter1.value, 0);
+		Assert.assertEquals(-5.2f, counter1.min, 0);
+		Assert.assertEquals(-3.0f, counter1.max, 0);
+		counter1.reset();
+		Assert.assertEquals(0, counter1.count);
+		Assert.assertEquals(0, counter1.total, 0);
+		Assert.assertEquals(Float.MAX_VALUE, counter1.min, 0);
+		Assert.assertEquals(-Float.MAX_VALUE, counter1.max, 0);
+		Assert.assertEquals(0, counter1.average, 0);
+		Assert.assertEquals(0, counter1.latest, 0);
+		Assert.assertEquals(0, counter1.value, 0);
+		Assert.assertEquals(0, counter1.mean.getMean(), 0);
+		Assert.assertEquals(0, counter1.mean.getValueCount(), 0);
+		Assert.assertEquals(0, counter1.mean.getLatest(), 0);
+		Assert.assertEquals(5, counter1.mean.getWindowSize());
+		Assert.assertArrayEquals(new float[] {}, counter1.mean.getWindowValues(), 0);
+
+		// Tests without a Windowed mean
+		counter2.reset();
+		Assert.assertNull(counter2.mean);
+		Assert.assertEquals(0, counter2.count);
+		Assert.assertEquals(0.0f, counter2.total, 0);
+		Assert.assertEquals(Float.MAX_VALUE, counter2.min, 0);
+		Assert.assertEquals(-Float.MAX_VALUE, counter2.max, 0);
+		Assert.assertEquals(0.0f, counter2.average, 0);
+		Assert.assertEquals(0.0f, counter2.latest, 0);
+		Assert.assertEquals(0.0f, counter2.value, 0);
+		counter2.put(1.0f);
+		Assert.assertEquals(1.0f, counter2.latest, 0);
+		Assert.assertEquals(1.0f, counter2.total, 0);
+		Assert.assertEquals(1.0f, counter2.average, 0);
+		Assert.assertEquals(1.0f, counter2.value, 0);
+		Assert.assertEquals(1, counter2.count);
+		Assert.assertEquals(1.0f, counter2.max, 0);
+		Assert.assertEquals(1.0f, counter2.min, 0);
+		counter2.put(34.0f);
+		counter2.put(15.24f);
+		counter2.put(43.62f);
+		Assert.assertEquals(43.62f, counter2.latest, 0);
+		Assert.assertEquals(93.86f, counter2.total, 0);
+		Assert.assertEquals(23.465f, counter2.average, 0);
+		Assert.assertEquals(43.62f, counter2.value, 0);
+		Assert.assertEquals(4, counter2.count);
+		Assert.assertEquals(43.62f, counter2.max, 0);
+		Assert.assertEquals(1.0f, counter2.min, 0);
+	}
+
+	@Test
+	public void testLimitCases () {
+		// Tests with a Windowed mean
+		counter1.reset();
+		counter1.put(Float.MIN_NORMAL);
+		Assert.assertEquals(1.17549435E-38f, counter1.latest, 0);
+		counter1.put(10000.85f);
+		Assert.assertEquals(10000.850000000000000000000000000000000000117549435f, counter1.total, 0);
+		counter1.put(75.15f);
+		counter1.put(7E18f);
+		Assert.assertEquals(1750000000000002519.00000000000000000000000000000000000002938735875f, counter1.average, 0);
+		counter1.put(5f);
+		Assert.assertEquals(1400000000000002013.80000000000000000000000000000000000002350988700f, counter1.mean.getMean(), 0);
```

Comment:
This looks like you've hardcoded values that will pass the test. Consider using `MathUtils#isEqual`.

Reply:
okay, will do !